### ISSUE TITLE
Feature: Allow Multiple instances of SquareDesk

### ISSUE THREAD

**Issue by Gero5** [mpogue2/SquareDesk#912]
When I start SquareDesk on a my second computer and did not close it on first one the following warning appears:

![Bildschirmfoto 2023-08-05 um 11 44 49](https://github.com/mpogue2/SquareDesk/assets/128936259/4b5763c7-a898-4eea-953c-846eb4c16552)

New feature: Implement a database solution that can be shared between multiple instances of SquareDesk so user does not have to take care of this.

**Comment by mpogue2** [comment]
This one is probably not possible to implement, given the use case (that we are both using):

- sqlite database instances on each machine
- cloud storage provider doing replication across multiple machines

For this use case, it's not possible to have two separate machines, each sharing an iCloud/Dropbox/Box.net COPY of the same database.  This is because iCloud/Dropbox/Box.net each use their own **non-sqlite-aware** replication strategy to duplicate the databases to each participating machine.  Those strategies can make copies and do overwriting at any time, and they do not let us know when they are updating/copying/overwriting.  There is no way that I'm aware of to lock the database to prevent the cloud storage providers from propagating the changes.

The warning is there so that the user knows that "this machine is taking over the database" can result in changes from the other machines being lost (remember: each machine is actually working on its own **local** copy).  In 99.9% of the use cases, we are working on only one machine at a time, and this is not a problem, because there's no simultaneous replication going on.  But, the warning is needed to know that this can happen if the user is not careful.

It would be possible to run a database server in the cloud to get around this problem (storing **everything** in the Cloud), but the simple cloud storage replication that we are using to copy around local database instances has the current limitations.

**Comment by danlyke** [comment]
A think I have thought about is that since the database is small enough to
store in memory, we could stash the whole thing in a text file and use git
to sync it, but we'd have to both serialize it sanely, and automatically
detect and understand git collisions (which would often result from plays
on multiple machines before the repo sync, but could result from adding
music files on independent machines).

But that starts to get into data structures and real computer science
stuff...

**Comment by mpogue2** [comment]
@danlyke surprisingly, I actually have commercial experience with an architecture that used git for syncing.  Use case was quite similar to ours, with multiple reader/writers that could be disconnected from the internet at any time.  I did not design it, I inherited it, but it was a huge failure (because it was extremely fragile).  My solution was to rearchitect with a proper database backend designed to avoid locks, which turned out to be quite robust.
